### DISCUSSION TITLE
Support WEBP images

### DISCUSSION THREAD

**Discussion by DavidGP** [pdf-association/pdf-issues#248] [Ideas]
The WEBP format offers much [superior compression](https://insanelab.com/blog/web-development/webp-web-design-vs-jpeg-gif-png/) over JPEG, JPEG2000 and even PNG, while additionally supporting transparency.

Are there any plans already when WEBP will be included in the PDF specification?

**Comment by petervwyatt** [comment]
ISO TC 171 SC 2 working groups have been researching the addition of new 2D image codecs into PDF for quite a while, with consideration given to various codecs (not file formats) to achieve smooth integration, supporting the advanced PDF transparent imaging model, the wide variety of PDF color spaces (incl. possible future support for HDR), IP, consideration for HW, meeting the needs of industry verticals (such as graphic arts and long-term preservation), etc. From an original long list of possibilities which included WebP with VP8 compression, the research of the working groups is now focused on AVIF with AV1 and JPEG-XT. This work still has a way to go but, no, WebP will not be supported in PDF in the foreseeable future.

**Reply by AlphaJack** [reply]
Are there updates regarding AVIF support?

**Reply by petervwyatt** [reply]
Yes - we have an internal TWG actively working on this but please understand that this also goes to HDR which then goes to color management, alpha compositing, transparency blending, etc etc. If you wish to participate please join our [Imaging Model TWG](https://pdfa.org/community/imaging-model-twg/).

**Reply by Peque** [reply]
@petervwyatt Out of curiosity, and knowing that whatever you say is in no way a commitment nor official position (just a personal opition): how long could it take until AVIF is part of PDF standard?

A very rough estimate with the order of magnitude would be enough (1 year, 2 years, 4...). :smile:

**Reply by petervwyatt** [reply]
Current discussions centre around adding [JPEG-XL](https://jpegxl.info/) support to PDF rather than AVIF or WEBP, with the [most recent discussions being in Prague](https://pdfa.org/event/pdf-week-prague-2024/). But there is a lot of work happening in other related areas (e.g. ISO TC 42 WG 23 with HDR gain maps (ISO 22028, ISO 21496); new CICP tags in ICC.1 profiles) so it is difficult to say when things will settle down sufficiently for stable adoption - my guess(!!) would be >=2 years for a formal ISO publication, as there is still a lot of technical issues to be resolved.

**Reply by Peque** [reply]
@petervwyatt Thanks! :blush: 

I hope you reconsider at least AVIF at some point. The importance of either WebP or AVIF would come from the use case of PDF generation from websites. This is widely used for generating reports, invoices, documents... Specially nowadays, when there is an increasingly number of SaaS that are web-based and allow some kind of PDF generation to download data. Basically, the source is in HTML, allowing to print as PDF with sometimes only some little CSS tweaks.

Both WebP and AVIF are widely supported across major browsers:

- https://caniuse.com/webp
- https://caniuse.com/avif

While JPEG-XL is not (and probably won't be, since Chrome/Chromium, in which many other browsers are based on, already [stated that will not be supporting it](https://issues.chromium.org/issues/40168998#comment85)):

- https://caniuse.com/jpegxl

This means that many websites out there are using WebP and/or AVIF already as image formats for good quality/compression, but when printing those pages, the generated PDF would convert them to PNG, making the file size huge and losing the originally crafted compression level. Even if [Skia](https://skia.org/) (or any other renderer) was updated to re-encode WebP/AVIF images instead of converting them to PNG, that would still mean image quality loss (plus, usually the already-available images are encoded with the right/expected quality/compression, while re-encoding would mean having to establish arbitrary values that may not match the original).

Have you considered this use case?

I'd say there are probably billions of web users that, at some point, have needed and will need to just "print as PDF" when surfing the web. :blush:

**Reply by petervwyatt** [reply]
The members of the [Imaging Model Technical Working Group](https://pdfa.org/community/imaging-model-twg/) considered various codecs/formats and have agreed on JPEG-XL as the primary way forward at this time.

**Reply by Peque** [reply]
Thanks @petervwyatt, I hope the use case I mentioned above is considered for future updates. The World Wide Web will be thankful for that. :smile:
